MahApps.Metro/Controls/RangeSlider.cs

```diff
@@ -2020,22 +2020,22 @@ private void SnapToTickHandle(ButtonType type, Direction mDirection, Double diff
         }
 
         //Calculating next value for Tick
-        private Double CalculateNextTick(Direction dir, double chekingValue, double distance, bool moveDirectlyToNextTick)
+        private Double CalculateNextTick(Direction dir, double checkingValue, double distance, bool moveDirectlyToNextTick)
         {
+            double checkingValuePos = checkingValue - Minimum;
             if (!IsMoveToPointEnabled)
             {
                 //Check if current value is exactly Tick value or it situated between Ticks
-                if (!IsDoubleCloseToInt((chekingValue - Minimum) / TickFrequency))
+                double checkingValueChanged = checkingValuePos + distance;
+                double x = checkingValueChanged / TickFrequency;
+                if (!IsDoubleCloseToInt(x))
                 {
-                    double x = (chekingValue - Minimum) / TickFrequency;
                     distance = TickFrequency * (int)x;
                     if (dir == Direction.Increase)
                     {
                         distance += TickFrequency;
                     }
-                    distance = (distance - Math.Abs(chekingValue - Minimum));
-                    _currenValue = 0;
-                    return Math.Abs(distance);
```

Comment:
@drayde i'm sure that we need this

without returning (the slider has a 20 tick frequency)
![2015-07-01 22h28_03](https://cloud.githubusercontent.com/assets/658431/8464194/b5b4ca42-2040-11e5-998b-01d395f39042.gif)

with returning, which works correct
![2015-07-01 22h30_01](https://cloud.githubusercontent.com/assets/658431/8464215/dc34f57a-2040-11e5-9bd3-f5b3e48ce5fa.gif)